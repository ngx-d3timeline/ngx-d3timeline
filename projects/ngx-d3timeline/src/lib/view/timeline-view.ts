export class TimelineView {
  readonly width: number;
  readonly height: number;
  readonly margin: number;

  readonly bottom: number;
  readonly left: number;
  readonly right: number;
  readonly top: number;

  constructor({
    width,
    height,
    margin
  }: {
    width: number;
    height: number;
    margin: number;
  }) {
    this.width = width;
    this.height = height;
    this.margin = margin;

    this.bottom = height - margin;
    this.left = 0;
    this.right = width - margin;
    this.top = 0;
  }
}
