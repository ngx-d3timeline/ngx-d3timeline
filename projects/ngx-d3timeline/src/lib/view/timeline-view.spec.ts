import { TimelineView } from './timeline-view';

describe('TimelineView', () => {
  it('should return correct bounds', () => {
    const timelineView = new TimelineView({
      width: 300,
      height: 400,
      margin: 50
    });

    expect(timelineView.bounds).toMatchSnapshot();
  });
});
