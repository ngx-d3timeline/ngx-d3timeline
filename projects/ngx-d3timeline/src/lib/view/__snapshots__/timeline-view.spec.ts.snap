// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineView should return correct bounds 1`] = `
TimelineView {
  "bottom": 350,
  "height": 400,
  "left": 0,
  "margin": 50,
  "right": 250,
  "top": 0,
  "width": 300,
}
`;
